db/model/Gdoc/rawToEnriched.ts

```diff
@@ -957,6 +931,182 @@ function parseCallout(raw: RawBlockCallout): EnrichedBlockCallout {
+    const textOnlyContent = raw.value.content.filter(
+        (element) => element.type === "text"
+    )
+    if (raw.value.content.length !== textOnlyContent.length) {
+        contentErrors.push({
+            message:
+                "Only paragraphs are supported in topic-page-intro blocks.",
+            isWarning: true,
+        })
+    }
+
+    const downloadButton = raw.value["download-button"]
+    if (downloadButton) {
+        if (!downloadButton.text) {
+            return createError({
+                message: "Download button specified but missing text value",
+            })
+        }
+
+        if (!downloadButton.url) {
+            return createError({
+                message: "Download button specified but missing url value",
+            })
+        }
+    }
+
+    const enrichedDownloadButton: EnrichedBlockTopicPageIntro["downloadButton"] =
+        downloadButton
+            ? {
+                  ...downloadButton,
+                  type: "topic-page-intro-download-button",
+              }
+            : undefined
+
+    const relatedTopics = raw.value["related-topics"]
+    const enrichedRelatedTopics: EnrichedTopicPageIntroRelatedTopic[] = []
+    if (relatedTopics) {
+        for (const relatedTopic of relatedTopics) {
+            if (!relatedTopic.url) {
+                return createError({
+                    message: "A related topic is missing a url",
+                })
+            }
+
+            const url = extractUrl(relatedTopic.url)
+            const { isGoogleDoc } = Url.fromURL(relatedTopic.url)
+            if (!isGoogleDoc && !relatedTopic.text) {
+                return createError({
+                    message:
+                        "A title must be provided for related topics that aren't linked via Gdocs",
+                })
+            }
+
+            // If we've validated that it's a Gdoc link without a title,
+            // or a regular link *with* a title, then we're good to go
+            const enrichedRelatedTopic: EnrichedTopicPageIntroRelatedTopic = {
+                type: "topic-page-intro-related-topic",
+                url,
+                text: relatedTopic.text,
+            }
+
+            enrichedRelatedTopics.push(enrichedRelatedTopic)
+        }
+    }
+
+    return {
+        type: "topic-page-intro",
+        downloadButton: enrichedDownloadButton,
+        relatedTopics: enrichedRelatedTopics,
+        content: textOnlyContent.map((rawText) =>
+            htmlToEnrichedTextBlock(rawText.value)
+        ),
+        parseErrors: [...contentErrors],
+    }
+}
+
+function parseKeyInsights(raw: RawBlockKeyInsights): EnrichedBlockKeyInsights {
+    const createError = (error: ParseError): EnrichedBlockKeyInsights => ({
+        type: "key-insights",
+        parseErrors: [error],
+        heading: "",
+        insights: [],
+    })
+
+    if (!raw.value.insights?.length) {
+        return createError({ message: "No insights included" })
+    }
```

Comment:
If I create a block that has no insights then instead of rendering a single block with an error I get no content and a document level error in the settings - is this new? It's not very nice in terms of writing content incrementally. I'd prefer for errors that don't HAVE to be document level errors to just draw a pink box component.

Reply:
You mean with something like this?

```
{.key-insights}
[.insights]
[]
{}
```

That renders the pink box but the rest of the content in the page renders fine.

![image](https://github.com/owid/owid-grapher/assets/11844404/20242cf9-dedb-4aaa-8228-f9505fbb9644)

I added content errors to the settings drawer also because when the article is long, it's annoying to have to scroll up and down looking for what the error is. 

Could you elaborate on what you mean by "writing content incrementally"? Other than the linking to unpublished docs (which are warnings, and thus not affected by this) I figured we wouldn't ever want authors to publish a topic page with parse errors. 🤔

Reply:
See https://ptolemy.owid.cloud/admin/gdocs/1M-sm24fryMd2JvGt-duQVUCHZN8GQD3VOTSx5YSjj_s/preview rn

Reply:
Okay that should be fixed now. It was due to `generateStickyNav` and `filterValidLinks` which is why I didn't get the same result as I was testing it on a different page that didn't have `type: topic-page`

Will proceed with the ts-pattern implementation now!